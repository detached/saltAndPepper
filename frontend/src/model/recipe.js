export class Recipe {
    id;
    author;
    title;
    category;
    cuisine;
    yields;
    ingredients;
    instructions;
    modifications;
    images;

    constructor(title, category, cuisine, yields, ingredients, instructions, modifications) {
        this.title = title;
        this.category = category;
        this.cuisine = cuisine;
        this.yields = yields;
        this.ingredients = ingredients;
        this.instructions = instructions;
        this.modifications = modifications;
    }

    addNewImages(images) {
        if (this.newImages === undefined) {
            this.newImages = [];
        }
        this.newImages = this.newImages.concat(images);
    }
    removeNewImage(imageFile) {
        if (this.newImages === undefined) {
            return
        }
        this.newImages = this.newImages.filter((image) => image !== imageFile);
    }

    removeExistingImage(imageId) {
        this.images = this.images.filter((image) => image.id !== imageId);
    }
}

export class RecipeImage {
    id;
    url;
    thumbnailUrl;

    constructor(id, url, thumbnailUrl) {
        this.id = id;
        this.url = url;
        this.thumbnailUrl = thumbnailUrl;
    }
}

export class Author {
    constructor(id, name) {
        this.id = id;
        this.name = name;
    }
}