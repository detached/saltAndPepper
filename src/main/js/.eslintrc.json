{
    "extends": [
        "eslint:recommended",
        "plugin:lit/recommended"
    ],
    "plugins": [
        "lit"
    ],
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "module"
    },
    "env": {
        "browser": true
    }
}