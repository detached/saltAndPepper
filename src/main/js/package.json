{
  "name": "recipe-js",
  "version": "1.0",
  "license": "UNLICENCED",
  "private": true,
  "scripts": {
    "build": "webpack --mode development"
  },
  "devDependencies": {
    "eslint": "^7.16.0",
    "eslint-plugin-lit": "^1.3.0",
    "eslint-webpack-plugin": "^2.4.1",
    "webpack": "^5.11.0",
    "webpack-cli": "^4.2.0"
  },
  "dependencies": {
    "lit-element": "^2.4.0"
  },
  "packageManager": "yarn@1.22.10"
}
