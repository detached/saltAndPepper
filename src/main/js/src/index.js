export {ImageGallery} from "./components/imageGallery";
export {copyToClipboard} from "./copyToClipboard";