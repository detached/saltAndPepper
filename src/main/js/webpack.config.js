const ESLintPlugin = require('eslint-webpack-plugin');

const path = require('path');

module.exports = {
  entry: './src/index.js',
  plugins: [new ESLintPlugin()],
  devtool: "source-map",
  output: {
    filename: 'bundle.js',
    path: path.resolve(__dirname, 'dist'),
    libraryTarget: 'umd'
  },
};
