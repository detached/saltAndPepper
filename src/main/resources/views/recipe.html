#parse("views/components/head.html")
<body>
<div id="layout">
    #parse("views/components/menu.html")
    <div id="main">
        <div class="header">
            <h1>$recipe.title</h1>
        </div>
        <div class="content centered">
            <image-gallery recipeId="$recipe.id"
                           toggleUploadButtonLabel="$translations.get('recipe.imageUpload')"
                           uploadImageLabel="$translations.get('image.upload')"
                           allowUpload=$recipe.allowedToEdit >
            </image-gallery>
            <div class="pure-g recipe-info">
                <div class="pure-u-1-2">$translations.get("recipe.author")</div>
                <div class="pure-u-1-2">$recipe.author</div>
                <div class="pure-u-1-2">$translations.get("recipe.category")</div>
                <div class="pure-u-1-2">$recipe.category</div>
                <div class="pure-u-1-2">$translations.get("recipe.cuisine")</div>
                <div class="pure-u-1-2">$recipe.cuisine</div>
                <div class="pure-u-1-2">$translations.get("recipe.yields")</div>
                <div class="pure-u-1-2">$recipe.yields</div>
            </div>
            <h2>$translations.get("recipe.ingredients")</h2>
            <p>
            #foreach($i in $recipe.ingredients.split("\n"))
            $i<br/>
            #end
            </p>

            <h2>$translations.get("recipe.instructions")</h2>
            <p>
            #foreach($i in $recipe.instructions.split("\n"))
            $i<br/>
            #end
            </p>

            <h2>$translations.get("recipe.modifications")</h2>
            <p>
            #foreach($i in $recipe.modifications.split("\n"))
            $i<br/>
            #end
            </p>

            #if ($recipe.allowedToEdit)
            <a class="pure-button pure-button-primary" href="$recipe.editUrl">
                $translations.get("recipe.edit")
            </a>
            <a class="pure-button recipe-delete-button" href="$recipe.deleteUrl">
                $translations.get("recipe.delete")
            </a>
            #end
        </div>
    </div>
</div>
</body>
#parse("views/components/foot.html")