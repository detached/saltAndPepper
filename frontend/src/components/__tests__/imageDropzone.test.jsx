import {render} from "@testing-library/react";
import ImageDropzone from "../imageDropzone.jsx";
import {RecipeImage} from "../../model/recipe.js";

describe("imageDropzone test", () => {
    test("when initialized with images, the images are shown in thumbnails", async () => {

        let initImages = [new RecipeImage({
            id: "1",
            url: "http://localhost/thumbnail",
            thumbnailUrl: "http://localhost/thumbnail"
        })];
        let onImagesDropped = () => {
        };
        let onImagesRemoved = () => {
        };

        let {container} = render(<ImageDropzone initImages={initImages} onImagesDropped={onImagesDropped}
                                                onImageRemoved={onImagesRemoved}/>);

        expect(container).toMatchSnapshot();
    })
});