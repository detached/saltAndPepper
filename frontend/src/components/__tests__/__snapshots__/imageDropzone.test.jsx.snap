// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`imageDropzone test > when initialized with images, the images are shown in thumbnails 1`] = `
<div>
  <section
    class="edit-recipe-images-container"
  >
    <div
      class="dropzone"
      role="presentation"
      tabindex="0"
    >
      <input
        accept="image/jpeg,.jpg,.jpeg,image/png,.png"
        multiple=""
        style="display: none;"
        tabindex="-1"
        type="file"
      />
      <p>
        Bilder hinzuf√ºgen
      </p>
    </div>
    <aside
      class="edit-recipe-images-thumbs-container"
    >
      <div
        class="edit-recipe-images-thumb"
      >
        <div>
          <img
            alt="thumbnail"
            id="image"
            src="http://localhost/thumbnail"
          />
          <div
            id="overlay"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
              />
            </svg>
          </div>
        </div>
      </div>
    </aside>
  </section>
</div>
`;
